solc/main.cpp

```diff
@@ -53,24 +52,48 @@ static void setDefaultOrCLocale()
 
 int main(int argc, char** argv)
 {
-	setDefaultOrCLocale();
-	solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
-	if (!cli.parseArguments(argc, argv))
+	try
+	{
+		setDefaultOrCLocale();
+		solidity::frontend::CommandLineInterface cli(cin, cout, cerr);
+		if (!cli.parseArguments(argc, argv) || !cli.readInputFiles() || !cli.processInput() || !cli.actOnInput())
+			return 1;
+
+		return 0;
+	}
+	catch (boost::exception const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	if (!cli.readInputFiles())
+	}
+	catch (Exception const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	if (!cli.processInput())
+	}
+	catch (InternalCompilerError const& _exception)
+	{
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
 		return 1;
-	bool success = false;
-	try
+	}
+	catch (smtutil::SMTLogicError const& _exception)
 	{
-		success = cli.actOnInput();
+		cerr << "Uncaught exception" << boost::diagnostic_information(_exception) << endl;
+		return 1;
 	}
-	catch (boost::exception const& _exception)
+	catch (std::exception const& _e)
 	{
-		cerr << "Exception during output generation: " << boost::diagnostic_information(_exception) << endl;
-		success = false;
+		cerr() << "Uncaught exception" << (_e.what() ? ": " + string(_e.what()) : ".") << endl;
+		return 1;
+	}
+	catch (Exception const& _exc)
```

Comment:
You need to import this exception type to be able to use it. It's defined in [`libsolutil/Exceptions.h`](https://github.com/ethereum/solidity/blob/develop/libsolutil/Exceptions.h).

Also, it inherits from both `std::exception` and `boost::exception` so it must be above them.